guides/source/form_helpers.md

```diff
@@ -867,15 +1012,31 @@ We can mix and match these two concepts. One element of a hash might be an array
 <input name="person[addresses][][city]" type="text"/>
 ```
 
-This would result in `params[:person][:addresses]` being an array of hashes with keys `line1`, `line2`, and `city`.
+This would result in `params[:person][:addresses]` being an array of hashes. Each hash in the array will have the keys `line1`, `line2`, and `city`, something like this:
 
-There's a restriction, however: while hashes can be nested arbitrarily, only one level of "arrayness" is allowed. Arrays can usually be replaced by hashes; for example, instead of having an array of model objects, one can have a hash of model objects keyed by their id, an array index, or some other parameter.
+```ruby
+{ 'person' =>
+  { 'addresses' => [
+    { 'line1' => '1000 Fifth Avenue',
+      'line2' => '',
+      'city' => 'New York'
+    },
+    { 'line1' => 'Calle de Ruiz de Alarcón',
+      'line2' => '',
+      'city' => 'Madrid'
+    }
+    ]
+  }
+}
+```
+
+It's important to note that while hashes can be nested arbitrarily, only one level of "arrayness" is allowed. Arrays can usually be replaced by hashes. For example, instead of an array of model objects, you can have a hash of model objects keyed by their id or similar.
 
-WARNING: Array parameters do not play well with the `check_box` helper. According to the HTML specification unchecked checkboxes submit no value. However it is often convenient for a checkbox to always submit a value. The `check_box` helper fakes this by creating an auxiliary hidden input with the same name. If the checkbox is unchecked only the hidden input is submitted and if it is checked then both are submitted but the value submitted by the checkbox takes precedence.
+WARNING: Array parameters do not play well with the `check_box` helper. According to the HTML specification unchecked checkboxes submit no value. However it is often convenient for a checkbox to always submit a value. The `check_box` helper fakes this by creating an auxiliary hidden input with the same name. If the checkbox is unchecked only the hidden input is submitted. If it is checked then both are submitted but the value submitted by the checkbox takes precedence. There is a `include_hidden` option that can be set to `false` if you want to omit this hidden field. By default, this option is `true`.
 
 ### The `fields_for` Helper `:index` Option
 
-Let's say we want to render a form with a set of fields for each of a person's
+Let's say you want to render a form with a set of fields for each of a person's
 addresses. The [`fields_for`][] helper with its `:index` option can assist:
```

Comment:
This fields_for currently isn't linked:
<img width="821" alt="image" src="https://github.com/rails/rails/assets/28561/a4fa809f-d1e3-47f5-bf08-235b6df79bdf">

```suggestion
Let's say you want to render a form with a set of fields for each of a person's
addresses. The [`fields_for`][] helper with its `:index` option can assist:
[`fields_for`]: https://api.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html#method-i-fields_for
```